plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/server/LocalAppEngineServerLaunchConfigurationDelegate.java

```diff
@@ -101,4 +128,53 @@ private int getDebugPort() throws CoreException {
     }
     return port;
   }
+
+
+  /**
+   * Open a browser on the provided page on server start. We actually wait 200 ms in case the server
+   * is subsequently stopped.
+   */
+  private class OpenBrowserListener implements IServerListener {
+    private String pageLocation;
+    private int waitTime = 1500;
```

Comment:
Hmm, good point.  This is a real hack: the real problem is that `LocalAppEngineServerBehaviour`'s `LocalAppEngineStartListener` should not set STATE_STARTED until it's able to probe the devserver's API port.  Or maybe that responsibility should be with _appengine-plugins-core_.